### PR TITLE
store/tikv: remove use of SchemaLease transaction option in store/tikv

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
Usage of `SchemaChecker` option in `store/tikv` can be removed.

Part of https://github.com/pingcap/tidb/issues/24302

### What is changed and how it works?
What's Changed:
- add `schemaLeaseChecker` in `KVTxn`
- add method `SetSchemaLeaseChecker` to set up the checker
- update `driver.tikvTxn` to adapt.

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -1398,8 +1399,8 @@ func (c *twoPhaseCommitter) checkSchemaValid(ctx context.Context, checkTS uint64
 		err := errors.Errorf("mock check schema valid failure")
 		failpoint.Return(nil, false, err)
 	})
-	checker, ok := c.txn.us.GetOption(kv.SchemaChecker).(schemaLeaseChecker)
-	if !ok {
+	checker := c.txn.schemaLeaseChecker
+	if checker == nil {
```
Comment (AndreMouche) [inline store/tikv/2pc.go:1403]: how about REMOVE var `checker` and `if c.txn.schemaLeaseChecker== nil `

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]
LGTM. But just the existence of `SchemaLeaseCheck` in the client is weird after we moved the whole client out of TiDB. We can change it later :)

File: store/tikv/2pc.go (hunk 1 above)
Comment (disksing) [reply store/tikv/2pc.go:1403]: ok. Good idea.

**Comment by disksing** [comment]
> LGTM. But just the existence of `SchemaLeaseCheck` in the client is weird after we moved the whole client out of TiDB. We can change it later :)

Totally agree! I was also thinking the main subject here is the option so...

**Comment by tisonkun** [comment]
this title is `remove SchemaLease option` but actually it is about Transaction option `SchemaChecker`. Also we do not remove the option. I'm confused by the title though.

**Comment by disksing** [comment]
> this title is `remove SchemaLease option` but actually it is about Transaction option `SchemaChecker`. Also we do not remove the option. I'm confused by the title though.

Good point.

**Pull request merged**